tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -419,6 +419,14 @@ internal Span StartSpan(string operationName, ITags tags = null, ISpanContext pa
                 }
             }
 
+            if (SourceLinkTagsProvider.Instance.GetGitTagsFromSourceLink() is { } sourceLinkTags)
+            {
+                foreach (var entry in sourceLinkTags)
+                {
+                    span.SetTag(entry.Key, entry.Value);
+                }
+            }
+
```

Comment:
This will likely have quite an impact on throughput tests - might be worth enabling there to check. 

Also, this isn't great for perf because it's adding tags to _every_ span, _and_ taking the slow-path of using the dictionary.

I feel like we should move this so that the source link tags are added directly during serialization? That would avoid the many extra allocations?

Reply:
I think we should cache this as well and add the Tags to the [CommonTags](https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Tagging/CommonTags.cs)

Reply:
I decided against adding it to `CommonTags` because it would mean we'd be converting the value to bytes each time, even though these values should never change in the entire lifetime of the application.

Instead, opted to use the same approach as we currently use for `environment` and `version` as per #3316. 
Note that this means memory-usage-per-trace is increased as the size of the `TraceContext` and `TraceChunkModel` have are both larger now. We should perhaps explore refactoring this away to improve perf in the future, but I think doing this is out of scope for this PR. /cc @lucaspimentel